data/kz/4.0.dict

```diff
@@ -0,0 +1,135 @@
+%
+% Kazakh dictionary -- experimental -- 2016 -- Tatiana Batura, Aigerim Bakiyeva, Aigerim Yerimbetova 
+%
+% Dictionary version number is 5.3.0 (formatted as V5v3v0+)
+<dictionary-version-number>: V5v3v0+;
+% The empty word.
+EMPTY-WORD.zzz: ZZZ-;
+
+мен.pron: {S1s+};
+сен.pron: {S2s+};
+сіз.pron: {Sr2s+};
+ол.pron: {S3s+};
```

Comment:
I notice that you are using curly braces {} everywhere.  You realize that this means that the connector is optional?  That is,  the connector does not have to be used if its not there?    I believe that in most cases, you should not be using the curly braces, since you DO want to use these connectors.  

Thus, for the above, I believe that this would be more correct:
```
мен.pron: S1s+;
сен.pron: S2s+;
```
and so on ...

If a word cannot be connected, then the parser has a built-in mechanism for ignoring such words;  you do not need to make them optional to allow them to be ignored.